.github/workflows/build-container-images.yml

```diff
@@ -0,0 +1,280 @@
+    steps:
+      - id: cache-workflow-restore
+        name: Restore Checkout & Compile Workflows
+        uses: actions/cache/restore@v4
+        with:
+          fail-on-cache-miss: true
+          key: checkout-and-compile-workflow-${{ github.sha }}
+          path: |
+            ~/work/exchange-api/exchange-api/.git
+            ~/work/exchange-api/exchange-api/.github
+            ~/work/exchange-api/exchange-api/src/main/resources/version.txt
+
+      - name: Restore Checkout & Compile Deliverables
+        id: cache-deliverable-restore
+        uses: actions/cache/restore@v4
+        with:
+          fail-on-cache-miss: true
+          key: checkout-and-compile-deliverable-${{ github.sha }}
+          path: ~/work/exchange-api/exchange-api/target
+
+      - continue-on-error: true
+        id: exchange-version
+        name: Exchange Version
+        run: |
+          echo "version=$(head -n 1 target/scala-*/classes/version.txt)" >> $GITHUB_OUTPUT
+          echo "version-major=$(grep -Eo '^[[:digit:]]+' target/scala-*/classes/version.txt)" >> $GITHUB_OUTPUT
+          echo "version-less-patch=$(grep -Eo '^[[:digit:]]+\.[[:digit:]]+' target/scala-*/classes/version.txt)" >> $GITHUB_OUTPUT
+
+      # QEMU for software emulation of multiple platforms
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set Up Docker
+        uses: docker/setup-docker-action@v4
+        with:
+          daemon-config: |
+            {
+              "debug": true,
+              "features": {
+                "containerd-snapshotter": true
+              }
+            }
+
+      # Docker buildx/buildkit for multi-platform builds
+      #  "linux/amd64,linux/amd64/v2,linux/amd64/v3,linux/arm64,linux/riscv64,linux/ppc64le,linux/s390x,linux/386,linux/mips64le,linux/mips64,linux/loong64,linux/arm/v7,linux/arm/v6"
+      - name: Setup Docker Buildx
+        with:
+          install: true
+          platforms: ${{ matrix.build-targets == 'exchange-ubi' && env.IMAGE_PLATFORMS || format('{0},linux/riscv64', env.IMAGE_PLATFORMS) }}
+        uses: docker/setup-buildx-action@v3
+
+      # Authenticate Docker Hub
+      - name: Login to Docker Hub's Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_TOKEN }}
+
+      # Authenticate GHCR
+      - name: Login to Github Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # Authenticate Quay
+      - name: Login to Quay.io's Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USER }}
+          password: ${{ secrets.QUAY_TOKEN }}
```

Comment:
We could skip authentication if given target registry is not enabled.

Reply:
Yes, I forgot to add that in. Oversight.